### PR TITLE
Refactor server code

### PR DESCRIPTION
**Pull request by osherdp**

### CODE REVIEW

**Review by IamShobe — APPROVED** [review]
Seems good 👍 
So now gunicorn is hosting the server?
Two questions: 
why is the server running with debug mode?
and how does the port changes if u expose port  8000 staticly?

File: src/porfavor/client.py [hunk 1]
```diff
@@ -12,66 +12,93 @@
...
     Args:
-        host (str): URL for reaching the server, e.g.: "docs:5000/".
-        project (str): the project's name.
-        root_dir (str): path to the root directory containing the
-            documentation (should include a "index.html" file and such).
+        file_or_directory (str): path for the filesystem file or directory to
+            be zipped.
+
+    Returns:
+        str: bytes of the zip file containing required files.
```
Comment (IamShobe) [inline src/porfavor/client.py:31]: str.

File: src/porfavor/client.py [hunk 2]
```diff
@@ -12,66 +12,93 @@
...
+        content = zip_content(file_or_directory)
         click.secho("DONE!", bold=True, fg="green")
 
-    finally:
-        if temp_dir is not None:
-            shutil.rmtree(temp_dir)
+    click.secho("Publishing content for project '{}'... ".format(project),
+                nl=False)
+
+    filename = "{}.zip".format(project)
```
Comment (IamShobe) [inline src/porfavor/client.py:76]: i dont know how to select multiple lines but i think that all of the below should be splited into a separate function.

File: src/porfavor/client.py [hunk 3]
```diff
@@ -12,66 +12,93 @@
...
 @click.argument("host", metavar="<host>")
 @click.argument("project", metavar="<project>")
-@click.argument("directory", metavar="<directory>",
-                type=click.Path(exists=True, file_okay=False))
-def publish_cli(host, project, directory):
+@click.argument("file_or_directory", metavar="<file_or_directory>",
+                type=click.Path(exists=True))
+def publish_cli(host, project, file_or_directory):
     """Publish documentation for a project on the server."""
     publish(host=host,
```
Comment (IamShobe) [inline src/porfavor/client.py:111]: i dont really see a reason for this split - its a function that calls another function and thats it.
better be splittingg the ffunction itself to logic parts and have an high abstraction level to read.

File: src/porfavor/server/api.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        }
+
+    return Response(json.dumps(projects), mimetype="application/json")
+
+
+@api.route('/upload_docs', methods=["PUT"])
+def upload_documentation():
+    """Enable deployment of documentation files."""
+    if not request.files:
+        return "No file was supplied in the request", 400
```
Comment (IamShobe) [inline src/porfavor/server/api.py]: change return code to be used from httplib. this way is more readable

Comment (IamShobe) [reply src/porfavor/server/api.py]: here and everywhere else

File: src/porfavor/server/main.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+    return render_template("index.html",
+                           projects=projects)
+
+
+@app.route('/static/<path:path>')
+def static_serve(path):
+    """Serve static files."""
+    static_directory = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)),
```
Comment (IamShobe) [inline src/porfavor/server/main.py]: that can   fit in 1 line no??

**Review by IamShobe — CHANGES_REQUESTED** [review]
LOL Until now i didnt know i  needed to click  on finish review XDD

File: src/porfavor/client.py (hunk 1 above)
Comment (osherdp) [reply src/porfavor/client.py:31]: Actually, the convention should be with a colon:
[https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)
(Search for "Returns:")

**Pull request merged**